libs/Smarty_Compiler.class.php

```diff
@@ -556,7 +563,7 @@ function _compile_tag($template_tag)
 
             case 'php':
                 /* handle folded tags replaced by {php} */
-                list(, $block) = each($this->_folded_blocks);
+                $block = array_shift($this->_folded_blocks);
```

Comment:
`array_shift` is intrusive, it will remove an element from `_folded_blocks` which is very different from just fetching it.
I think that
```
$block = current($this->_folded_blocks);
next($this->_folded_blocks);
```
will be better